src/mito2/src/sst/index/codec.rs

```diff
@@ -62,9 +64,9 @@ impl IndexValueCodec {
 pub struct IndexValuesCodec {
     /// Tuples containing column id and its corresponding index_name (result of `to_string` on ColumnId),
     /// to minimize redundant `to_string` calls.
-    column_ids: Vec<(ColumnId, String)>,
+    column_ids: HashMap<ColumnId, String>,
     /// The data types of tag columns.
-    fields: Vec<SortField>,
+    fields: Vec<(ColumnId, SortField)>,
     /// The decoder for the primary key.
     decoder: DensePrimaryKeyCodec,
```

Comment:
Do you have plan to replace this with `Arc<dync PrimaryKeycodec>`?

Reply:
Good catch